pkg/controller/runtime/runtime.go

```diff
@@ -36,6 +36,8 @@ type Runtime struct { //nolint:govet
 	watchedMu   sync.Mutex
 	watched     map[watchKey]struct{}
 
+	trackingOutputPool trackingOutputPool
```

Comment:
I would actually make it a global variable, since `sync.Pool` is thread safe. The reason for making it a global variable - it will amortize allocations across all controllers as opposed to being "per controller allocator".

Reply:
this is not per-controller! it's per-controller runtime

Reply:
I would prefer to keep runtimes isolated, and no global vars. Usual case is to have a single controller-runtime per process, so it's effectively shared pool.